### ISSUE TITLE
Change RTMP url in pull dynamically?

### ISSUE THREAD

**Issue by KCastiglione** [arut/nginx-rtmp-module#667]
Is it possible to change RTMP url in dynamically either using control module or any other way to do it?
It will be able to great to do that!

        application pull {
            live on;
            pull rtmp://XXXXX/live;
        }

**Comment by misiek08** [comment]
Try with $name and $app. Those are documented in wiki, you need to look for the variables, because I don't remember exact names.

**Comment by KCastiglione** [comment]
@misiek08 is it possible to change the actual URL rtmp://XXXXXX - the XXXXX part .

**Comment by songshaodong** [comment]
@KCastiglione  good idea , the workload will be relatively large, because now completely static way.

**Comment by gabrielfarah** [comment]
Hi, any update on this issue? Or a way it could be achieved?

**Comment by JoeWoodward** [comment]
This is exactly what I'm looking for too. Currently I need to have multiple applications for each master node however I'd like to pass the url as a query parameter and use that variable in the pull directive i.e.

```
application watch {
  live on;
  pull rtmp://$args_ip/ingress_app/$name live=1;
}
```

This I can send my viewers to `rtmp://ip_of_edge_node/watch/channel_name?ip=ip_of_master_node` which would dynamically pull from the correct source

**Comment by semihyonet** [comment]
Hey, are there any updates on this issue?
